main.cc

```diff
@@ -1724,11 +1724,6 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             auto stop_repair_service = defer_verbose_shutdown("repair service", [&repair] {
                 repair.stop().get();
             });
-            repair.invoke_on_all(&repair_service::start).get();
-            api::set_server_repair(ctx, repair).get();
-            auto stop_repair_api = defer_verbose_shutdown("repair API", [&ctx] {
-                api::unset_server_repair(ctx).get();
-            });
```

Comment:
Does repair_service have an implicit dependency on batchlog_manager? If so it should be made explicit (passed via the constructor)..

Reply:
It has explicit dependency. This place calls repair_service::start() on all shards, but before that there's sharded<repair_service>::start() call that's not moved and it has batchlog manager argument passed to repair_service constructor